### DISCUSSION TITLE
Renewing certificate of an entra application

### DISCUSSION THREAD

**Discussion by stevebeauge** [pnp/powershell#4551] [Q&A]
PNP.Powershell provides the ability to register ENTRA applications using Register-PnPEntraIDApp (for app only authentification).

This will provision an entra application and a certificate to handle the authentication.

How can I handle certificate expiration ? Is there any similar command to renew the certificate?

Thanks

**Comment by LeonArmston** [comment]
I have the same ask - I'd like to re-generate the certificate for an existing app registration I created using Register-PnPEntraIDApp - I dont believe this is possible using PnP PowerShell?

Is it possible using PowerShell?

**Reply by LeonArmston** [reply]
Not to worry I resolved it using #[New-PnPAzurecertificate](https://pnp.github.io/powershell/cmdlets/New-PnPAzureCertificate.html) and then when to the App Registration in Entra ID though the UI then to Certificates and then uploaded the **.cer** file.

Code below to generate the cert
```powershell
$certPassword = "PasswordHere"
$appName = "My App"
# Remove spaces and illegal filename characters from $certName for the filename
$sanitizedCertName = ($appName -replace '[\\\/\:\*\?\"<>\| ]', '')
New-PnPAzureCertificate -CommonName $certName -ValidYears 10 -Store CurrentUser -OutCert "c:\temp\$($sanitizedCertName).cer" -OutPfx "c:\temp\$($sanitizedCertName).pfx" -CertificatePassword (ConvertTo-SecureString -String $certPassword -AsPlainText -Force)
```
